mujoco_warp/_src/io.py

```diff
@@ -516,6 +520,17 @@ def _constraint(mjm: mujoco.MjModel, nworld: int, njmax: int) -> types.Constrain
     shape=(nworld, mjm.opt.ls_iterations), dtype=wp.vec3f
   )
 
+  # TODO(team): skip allocation if not elliptic?
+  efc.fri = wp.empty((nconmax, 6), dtype=wp.float32)
+  efc.dm = wp.empty((nconmax,), dtype=wp.float32)
```

Comment:
We seem to be storing a lot of temporaries that in C MuJoCo are calculated inline and then discarded.

Presumably this is because you have lots of kernels with different launch dims and you need to store intermediate results.

Are there ways to fuse those kernels so you don't need all these extra arrays?

Reply:
@erikfrey
- a number of the variables for elliptic friction cones have been removed from `Constraint` and are now calculated inline
- `_cone_hessian` has been removed
- `_friction_elliptic` has been removed
- `_efc_elliptic1` and `_efc_elliptic2` are combined into a single kernel

Reply:
Amazing - great work.  No perf impact?